pkg/updatestatus/start.go

```diff
@@ -0,0 +1,49 @@
+package updatestatus
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/informers"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+)
+
+const (
+	uscNamespace = "openshift-cluster-version"
+)
+
+// Run starts all update status controllers
+func Run(ctx context.Context, cc *controllercmd.ControllerContext) error {
+	configClient, err := configv1client.NewForConfig(cc.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	coreClient, err := kubeclient.NewForConfig(cc.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	uscNamespaceCoreInformers := informers.NewSharedInformerFactoryWithOptions(coreClient, 10*time.Minute, informers.WithNamespace(uscNamespace))
+
+	updateStatusController, sendInsight := newUpdateStatusController(ctx, coreClient, uscNamespaceCoreInformers, cc.EventRecorder)
+	controlPlaneInformerController := newControlPlaneInformerController(configInformers, cc.EventRecorder, sendInsight)
+
+	configInformers.Start(ctx.Done())
+	uscNamespaceCoreInformers.Start(ctx.Done())
+
+	go updateStatusController.Run(ctx, 1)
+	go controlPlaneInformerController.Run(ctx, 1)
```

Comment:
The CVO has [a rather elaborate system][1] to track child goroutines, wait some time for them to gracefully terminate, and collect `error` output.  We may not need that kind of graceful shutdown and exit code handling in this controller?  Because we're hopping to kick out `controlPlaneInformerController` soon, after which point we can drop the goroutine for `updateStatusController`?

[1]: https://github.com/openshift/cluster-version-operator/blob/adf0cf5662854567d1ea7ce1590e0a6fed8fcaaa/pkg/cvo/cvo.go#L418-L515

Reply:
TBH I'm not entirely sure - I followed some existing operators using library-go code the ones I found used this pattern without any further logic:

- https://github.com/openshift/service-ca-operator/blob/1f7d6472fea332fc307d9c8911f8c3cf57347755/pkg/operator/starter.go#L117-L137
- https://github.com/openshift/cluster-authentication-operator/blob/e401c0f1fbf56c7fa957daf5a8a8c91e568c646f/pkg/operator/starter.go#L88-L92